include/envoy/upstream/upstream.h

```diff
@@ -301,6 +302,12 @@ class Cluster : public virtual HostSet {
   virtual ClusterInfoConstSharedPtr info() const PURE;
 
   /**
+   * @return a shared pointer to the cluster's outlier detector. If an outlier detector has not been
+   *         installed, returns a nullptr.
+   */
+  virtual const Outlier::DetectorSharedPtr outlierDetector() const PURE;
```

Comment:
Ah I see this is a const pointer not a pointer to a const object. What we want is to return a pointer to a const object that way we can't change the object through the pointer.